src/main/java/org/jooq/lambda/SeqBuffer.java

```diff
@@ -57,6 +63,17 @@ private SeqBuffer(Spliterator<T> source) {
         return Seq.seq(new BufferSpliterator());
     }
 
+    private static <T> List<T> initBuffer(Spliterator<T> source) {
+        long estimateSize = source.estimateSize();
+        if (estimateSize == Long.MAX_VALUE) // infinite/unknown size
+            return new ArrayList<>();
+
+        if (estimateSize > MAX_ARRAY_SIZE)
+            throw new IllegalArgumentException("Stream is too long to be buffered: " + estimateSize);
+
+        return new ArrayList<>((int) estimateSize);
```

Comment:
Ha, you're right! I'll revert it.

PS. We wouldn't be entirely back to square one because I mainly wanted to avoid up-front processor consumption (i.e. consuming entire `Seq`) but you're absolutely right that it's best to also avoid up-front memory consumption.